Monika-After-Story/MonikaModDev
# Better holdme

I added an advanced pause displayable. It supports custom events and hotkeys. A bit refactored all the places where you can hold Monika. With this we don't have to force no music anymore, and it makes holdme a bit more convenient. Technically this opens the door to changing time/weather during holding too.

Additional changes:
- time for Monika to fall asleep is randomized, and it'll take longer if you already hold her
- teasing Monika isn't guaranteed, and she's less likely to ask for another hug if you already hold her
- all holding related labels were refactored quite a bit
- reduced the speed RenPy memory leaks (but it's not fixed) for the new pause displayable
- `persistent._mas_last_hold` is now a `datetime.datetime` object
- The `monika_holdrequest` event always exists now. The default chance to get it is 20%, increasing every hour (25%, 33%, 50%, 100%)

Copy/paste for testing
```python
events = (
    PauseDisplayableEvent(
        datetime.timedelta(seconds=5),
        renpy.partial(renpy.invoke_in_new_context, renpy.say, store.m, "5 sec ev"),
        repeatable=True
    ),
    PauseDisplayableEvent(
        datetime.timedelta(seconds=10),
        renpy.partial(renpy.say, store.m, "10 sec ev"),
        invoke_in_new_context=True
    ),
    PauseDisplayableEvent(
        datetime.timedelta(seconds=3),
        renpy.partial(renpy.invoke_in_new_context, renpy.say, store.m, "3 sec ev"),
        invoke_in_new_context=True
    ),
    PauseDisplayableEvent(
        datetime.timedelta(seconds=15),
        renpy.partial(renpy.invoke_in_new_context, renpy.say, store.m, "15 sec ev"),
        invoke_in_new_context=True
    )
)
disp = PauseDisplayableWithEvents(events=events)
disp.start()
```
### Testing:
- verify all places where you can hold Monika still work
  - the lullaby starts playing in the appropriate time
  - we have good fadein/fadeout 's
  - the holdme request event is working and has a good probability to trigger
  - added exp/dlg are good
- give some test to `PauseDisplayableWithEvents` and `PauseDisplayableEvent` - try different params
- test update scripts
- verify no new bugs lol
